src/decNumber/decNumber.c

```diff
@@ -604,9 +605,10 @@ decNumber * decNumberFromString(decNumber *dn, const char chars[],
       // if it was too long the exponent may have wrapped, so check
       // carefully and set it to a certain overflow if wrap possible
       if (c>=firstexp+9+1) {
-        if (c>firstexp+9+1 || *firstexp>'1') exponent=DECNUMMAXE*2;
+        if (c>firstexp+9+1 || *firstexp>'1') expa=DECNUMMAXE*2;
         // [up to 1999999999 is OK, for example 1E-1000000998]
         }
+      exponent=(int)((expa<<1)>>1);     // clear sign bit

```

Comment:
I guess I should use `Int` instead of `int`; also clearing the sign bit should not be necessary since expa will be `<INT_MAX`  at this point